File: app/src/main/java/com/myapplication/rick/packagedelivery/RangeType.java
package com.myapplication.rick.packagedelivery;

public enum RangeType {
    Even,
    Uneven,
    All;

    public static RangeType getRangeTypeFromNumber(int number) {
        if (number % 2 == 0) {
            return Even;
        } else {
            return Uneven;
        }
    }
}

File: app/src/main/java/com/myapplication/rick/packagedelivery/RangeType.kt
package com.myapplication.rick.packagedelivery

import android.os.Parcel
import android.os.Parcelable

enum class RangeType : Parcelable {
    Even,
    Uneven,
    All;

    override fun writeToParcel(parcel: Parcel, flags: Int) {
        parcel.writeString(toString())
    }

    override fun describeContents(): Int = 0

    companion object CREATOR : Parcelable.Creator<RangeType> {
        override fun createFromParcel(parcel: Parcel): RangeType =
                RangeType.valueOf(parcel.readString())

        override fun newArray(size: Int): Array<RangeType?> = arrayOfNulls(size)
    }
}
